web/base_viewer.js

```diff
@@ -1273,16 +1272,14 @@ class BaseViewer {
       console.error(`isPageVisible: "${pageNumber}" is not a valid page.`);
       return false;
     }
-    return this._getVisiblePages().views.some(function (view) {
-      return view.id === pageNumber;
-    });
+    return this._getVisiblePages().ids.has(pageNumber);
   }
 
   /**
    * @param {number} pageNumber
    */
   isPageCached(pageNumber) {
-    if (!this.pdfDocument || !this._buffer) {
```

Comment:
*Slightly unrelated to the rest of the patch, but:* Note that `this._buffer` will always be defined whenever `this.pdfDocument` is set.